scripts/alert_emails/send_emails.ts

```diff
@@ -196,24 +200,33 @@ async function setLastSnapshotNumber(
                   sentAt: admin.firestore.FieldValue.serverTimestamp(),
                 });
             })
-            .catch(err => {
-              console.log(err);
-              errorCount += 1;
+            .catch(async (err: CampaignMonitorError) => {
+              // if the email is invalid, move the document to the invalid collection
+              if (err.Code === CM_INVALID_EMAIL_ERROR_CODE && err.Message === CM_INVALID_EMAIL_MESSAGE) {
+                invalidEmailCount += 1;
+                const currentData = (await db.collection("alerts-subscriptions").doc(doc.id).get()).data();
+                if (currentData) {
+                  await db.collection("invalid-alerts-subscriptions").doc(doc.id).set(currentData)
+                  await db.collection("alerts-subscriptions").doc(doc.id).delete();
```

Comment:
Would you be up for using a WriteBatch so this is atomic? 😁  Something like (untested):

```
await db.batch()
    .set(db.collection("invalid-alerts-subscriptions").doc(doc.id), currentData)
    .delete(db.collection("alerts-subscriptions").doc(doc.id))
    .commit();
```

Reply:
though I guess it really doesn't matter much...  it's just a great use case for write batches. 😁